platforms/nuttx/src/px4/common/cpuload.cpp

```diff
@@ -145,14 +155,23 @@ void sched_note_stop(FAR struct tcb_s *tcb)
 
 void sched_note_suspend(FAR struct tcb_s *tcb)
 {
-
 	if (system_load.initialized) {
-		uint64_t new_time = hrt_absolute_time();
+		if (tcb->pid == 0) {
+			system_load.tasks[0].total_runtime += hrt_elapsed_time(&system_load.tasks[0].curr_start_time);
+			return;
+
+		} else {
+			if (idle_thread_only.load()) {
+				return;
+			}
+		}
+
+		for (auto &task : system_load.tasks) {
+			// Task ending its current scheduling run
+			if (task.valid && (task.curr_start_time > 0)
```

Comment:
So that the task enabling cpuload monitor (top or logger) won't get caught initially showing a large percentage.
When `cpuload_monitor_start()` is called it zeros all existing task curr_start_time and total_runtime.